src/components/CalendarDay.jsx

```diff
@@ -46,6 +48,16 @@ export default class CalendarDay extends React.Component {
     onDayMouseLeave(day, e);
   }
 
+  renderDay(day) {
+    const { renderDay } = this.props;
+
+    if (renderDay) {
+      const result = renderDay(day);
+      return typeof result === 'string' ? result : Children.only(result);
```

Comment:
@majapw @ljharb `Children.only` doesn't accept strings, so I had to work around it a bit.